lightclient/src/background.rs

```diff
@@ -43,59 +47,58 @@ pub enum FromSubxt {
         params: String,
         /// Channel used to send back the subscription ID if successful.
         sub_id: oneshot::Sender<MethodResponse>,
-        /// Channel used to send back the notifcations.
+        /// Channel used to send back the notifications.
         sender: mpsc::UnboundedSender<Box<RawValue>>,
+        /// The ID of the chain used to identify the chain.
+        chain_id: ChainId,
     },
 }
 
 /// Background task data.
-pub struct BackgroundTask {
+pub struct BackgroundTask<TPlatform: PlatformRef, TChain> {
     /// Smoldot light client implementation that leverages the exposed platform.
-    client: smoldot_light::Client<PlatformType>,
-    /// The ID of the chain used to identify the chain protocol (ie. substrate).
-    ///
-    /// Note: A single chain is supported for a client. This aligns with the subxt's
-    /// vision of the Client.
-    chain_id: ChainId,
-    /// Unique ID for RPC calls.
-    request_id: usize,
+    client: smoldot_light::Client<TPlatform, TChain>,
+    /// Unique IDs for RPC calls.
+    rpc_chain_id: UniqueChainID,
     /// Map the request ID of a RPC method to the frontend `Sender`.
-    requests: HashMap<usize, oneshot::Sender<MethodResponse>>,
+    requests: HashMap<(usize, smoldot_light::ChainId), oneshot::Sender<MethodResponse>>,
     /// Subscription calls first need to make a plain RPC method
     /// request to obtain the subscription ID.
     ///
     /// The RPC method request is made in the background and the response should
     /// not be sent back to the user.
     /// Map the request ID of a RPC method to the frontend `Sender`.
     id_to_subscription: HashMap<
-        usize,
+        (usize, smoldot_light::ChainId),
         (
             oneshot::Sender<MethodResponse>,
             mpsc::UnboundedSender<Box<RawValue>>,
         ),
     >,
     /// Map the subscription ID to the frontend `Sender`.
-    subscriptions: HashMap<usize, mpsc::UnboundedSender<Box<RawValue>>>,
+    ///
+    /// The subscription ID is entirely generated by the node (smoldot). Therefore, it is
+    /// possible for two distinct subscriptions of different chains to have the same subscription ID.
+    subscriptions: HashMap<(usize, smoldot_light::ChainId), mpsc::UnboundedSender<Box<RawValue>>>,
```

Comment:
Is subscription ID generation an monotonic increasing counter in smoldot? 

For instance in jsonrpsee it's a randomly generated such that you can't really assume that the subscription ID is unique but unlikely to happen.

Reply:
I believe to be so since the first subscription ID is zero, although Id have to double check.
I've had some issues initially where two different chains created one subscription each, then the smoldot would provide sub_id = 0 for both and the multiplexing forwarded all responses to one chain instead.